docs/updates/when-the-world-stops.md

```diff
@@ -0,0 +1,216 @@
+
+There are three characters in the story: the client, server, and engine.
+When the page pauses, the engine pauses the page, tells the server,
+and the server tells the client.
+
+This post is going to focus on the work the server does when the page pauses. The code included has been simplified a bit to remove some of the incidental complexity.
+
+---
+
+The [server][script.js] has several page lifecycle hooks that are called when
+something happens. `onDebuggerStatement` is one of these hooks!
+
+```js
+onDebuggerStatement: function (frame) {
+  return this._pauseAndRespond(frame, { type: "debuggerStatement" });
+}
+```
+
+The [server][script.js]'s main job is to tell the client that the page is paused.
+You can see below, that the `_paused` function is building a `packet`, this packet
+has all of the data the client needs to inform the programmer about the pause state.
+The `_paused` function also does one other interesting thing here as well. It starts,
+registering all of the relevant actors in the story in various "actor pools".
+The server registers all of the actors so that if the client has any follow up questions,
+the server will know who to follow up with.
+
+The server is like a detective, it wants to know who all the major witnesses are!
+
+
+```js
+_paused: function (frame) {
+  this._state = "paused";
+
+
+  this._pausePool = new ActorPool(this.conn);
+  this.conn.addActorPool(this._pausePool);
+
+  // Give children of the pause pool a quick link back to the
+  // thread...
+  this._pausePool.threadActor = this;
+  this._pauseActor = new PauseActor(this._pausePool);
+  this._pausePool.addActor(this._pauseActor);
+
+  // Update the list of frames.
+  let poppedFrames = this._updateFrames();
+
+  // Send off the paused packet and spin an event loop.
+  let packet = { from: this.actorID,
+                 type: "paused",
+                 actor: this._pauseActor.actorID };
+  if (frame) {
+    packet.frame = this._createFrameActor(frame).form();
+  }
+
+  if (poppedFrames) {
+    packet.poppedFrames = poppedFrames;
+  }
+
+  return packet;
+}
+```
+
+Lets continue digging into how the server builds the pause packet for the client.
+We already know quite a bit already. We know the `type` is "paused" and it has a `from` and `actor` field as a return address :). We also know that there are two types of frame fields: `frame` and `poppedFrames`. Lets focus on the `frame` field because it is simpler.
+
+We get the `frame` packet data by asking for the frame actor's `form`.
+The form function is defined in the Frame actor [class][frame.js].
+Lets slow down and take a look at the data the frame actor puts in its form.
+It's fascinating!
+
+It starts with the `actor` and `type` field for record keeping. It then sets up `callee`, `environment`, `arguments`, `where`, and `oldest` fields. Oldest, what's that? :) It's neat to think of a frame as potentially a function frame, at which point we care about the callee and arguments data.
```

Comment:
I don't feel like we're explaining what `oldest` is here, or at least I don't understand what it does.